cli/command/service/logs.go

```diff
@@ -92,7 +92,7 @@ func runLogs(dockerCli *command.DockerCli, opts *logsOptions) error {
 		if !client.IsErrServiceNotFound(err) {
 			return err
 		}
-		task, _, err := cli.TaskInspectWithRaw(ctx, opts.target)
+		task, _, _ := cli.TaskInspectWithRaw(ctx, opts.target)
```

Comment:
Instead of removing the err, do this:

```go
task, _, err := cli.TaskInspectWithRaw(ctx, opts.target)
if err != nil {
    if client.IsErrTaskNotFound(err) {
        // if the task isn't found, rewrite the error to be clear
        // that we looked for services AND tasks and found none
        err = fmt.Errorf("No such task or service")
    }
    return err
}
```

Basically, moving lines 106-113 to this point in the code.